mesh/mesh.go

```diff
@@ -92,18 +130,51 @@ func (m *Mesh) AddLayer(layer *Layer) error {
 		m.Debug("can't add layer", layer.Index(), " missing previous layers")
 		return errors.New("can't add layer missing previous layers")
 	}
-
+	atomic.StoreUint32(&m.lastSeenLayer, uint32(layer.Index()))
 	m.addLayer(layer)
-	m.tortoise.HandleIncomingLayer(layer)
-	atomic.StoreUint32(&m.localLayer, uint32(layer.Index()))
 	m.SetLatestLayer(uint32(layer.Index()))
 	return nil
 }
 
+func (m *Mesh) ValidateLayer(layer *Layer){
+	m.tortoise.HandleIncomingLayer(layer)
+}
+
+func (m *Mesh) LayerCompleteCallback(layerId LayerID){
```

Comment:
we don't really need the callbacks like we talked about
one can just look at the verifiedLayer index to see what finished and what diden't.